### PR TITLE
[DPEDE-911] Improve Chi Vue Toolbar for Vue 3 migration

### PR DESCRIPTION
**Pull request by josecarlosgonzalezv**
https://ctl.atlassian.net/browse/DPEDE-911

### CODE REVIEW

File: src/chi-vue/src/components/column-customization/ColumnCustomizationModalContent.tsx [hunk 1]
```diff
@@ -67,12 +67,12 @@ export default class ColumnCustomizationContent extends Vue {
...
-        ? 'Deselect'
-        : icon === 'chevron-right'
-        ? 'select'
-        : '';
+          ? 'Move Down'
+          : icon === 'chevron-left'
+            ? 'Deselect'
+            : icon === 'chevron-right'
+              ? 'select'
+              : '';
```
Comment (SergioQCL) [inline src/chi-vue/src/components/column-customization/ColumnCustomizationModalContent.tsx:75]: Nesting tertiary operators is not a good practice. Please, use a switch for this

File: src/chi-vue/src/components/data-table-filters/AdvancedFilters.tsx [hunk 2]
```diff
@@ -259,12 +259,12 @@ export default class AdvancedFilters extends Vue {
           filter.type === 'select'
             ? this._createSelectFilter(filter)
             : filter.type === 'input'
-            ? this._createInputFilter(filter)
-            : filter.type === 'checkbox'
-            ? this._createCheckboxFilter(filter)
-            : filter.type === 'textarea'
-            ? this._createTextareaFilter(filter)
-            : null;
+              ? this._createInputFilter(filter)
```
Comment (SergioQCL) [inline src/chi-vue/src/components/data-table-filters/AdvancedFilters.tsx:262]: Same here

File: src/chi-vue/src/components/data-table-filters/DataTableFilters.tsx [hunk 3]
```diff
@@ -336,23 +327,23 @@ export default class DataTableFilters extends Vue {
         filter.type === 'select'
           ? this._createSelectFilter(filter)
           : filter.type === 'input'
-          ? this._createInputFilter(filter)
-          : filter.type === 'checkbox'
-          ? this._createCheckboxFilter(filter)
-          : filter.type === 'textarea'
-          ? this._createTextareaFilter(filter)
-          : null;
+            ? this._createInputFilter(filter)
```
Comment (SergioQCL) [inline src/chi-vue/src/components/data-table-filters/DataTableFilters.tsx:330]: +1

File: src/chi-vue/src/components/data-table-filters/DataTableFilters.tsx [hunk 4]
```diff
@@ -336,23 +327,23 @@ export default class DataTableFilters extends Vue {
...
         filter.type === 'select'
           ? this._createSelectFilter(filter, true)
           : filter.type === 'input'
-          ? this._createInputFilter(filter, true)
-          : filter.type === 'checkbox'
-          ? this._createCheckboxFilter(filter, true)
-          : filter.type === 'textarea'
-          ? this._createTextareaFilter(filter, true)
-          : null;
+            ? this._createInputFilter(filter, true)
```
Comment (SergioQCL) [inline src/chi-vue/src/components/data-table-filters/DataTableFilters.tsx:341]: +1

File: src/chi-vue/src/components/data-table-toolbar/DataTableToolbar.vue
Comment (SergioQCL) [inline src/chi-vue/src/components/data-table-toolbar/DataTableToolbar.vue:1]: ❤️

File: src/chi-vue/src/components/data-table/DataTable.tsx [hunk 5]
```diff
@@ -891,8 +895,8 @@ export default class DataTable extends Vue {
       rowData && typeof rowData === 'object' && rowData.rowNumber
         ? `checkbox-select-${rowData?.rowId}`
         : selectAll
-        ? `checkbox-${this._dataTableId}-select-all-rows`
-        : '';
+          ? `checkbox-${this._dataTableId}-select-all-rows`
```
Comment (SergioQCL) [inline src/chi-vue/src/components/data-table/DataTable.tsx:898]: +1

File: src/chi-vue/src/components/data-table/DataTable.tsx [hunk 6]
```diff
@@ -1911,8 +1910,8 @@ export default class DataTable extends Vue {
         this.showSelectedOnlyRowsData?.length > 0
           ? this.showSelectedOnlyRowsData
           : this.sortedData && this.sortedData.length > 0
-          ? this.sortedData
-          : this.serializedDataBody;
+            ? this.sortedData
```
Comment (SergioQCL) [inline src/chi-vue/src/components/data-table/DataTable.tsx:1913]: +1

File: src/chi-vue/src/components/search-input/SearchInput.tsx [hunk 7]
```diff
@@ -41,8 +43,13 @@ export default class SearchInput extends Vue {
 
   @Emit(SEARCH_INPUT_EVENTS.SEARCH)
   _emitSearch(value: string) {
+    if (this.toolbarSearch && this.toolbarSearch.callback) {
```
Comment (SergioQCL) [inline src/chi-vue/src/components/search-input/SearchInput.tsx:46]:
```suggestion
    if (this.toolbarSearch?.callback) {
```

File: src/chi-vue/src/constants/constants.ts [hunk 8]
```diff
@@ -37,3 +37,9 @@ export const EPANEL = {
   DEFAULT_TOOLTIP_CONTENT: 'Done',
 };
 //#endregion
+
+//#region Toolbar keys
+export const TOOLBAR_KEYS = {
+  SEARCH_INPUT: Symbol('chi-toolbar-search'),
```
Comment (SergioQCL) [inline src/chi-vue/src/constants/constants.ts:43]: Just curious, why is the Symbol needed here?

File: src/chi-vue/src/views/DataTable/ClientSide/DataTableClientView.vue [hunk 9]
```diff
@@ -57,13 +57,7 @@
...
     </template>
     <template v-slot:toolbar>
-      <DataTableToolbar
-        @chiToolbarFiltersChange="(e) => chiToolbarFiltersChange(e)"
-        @chiToolbarSearch="(e) => chiToolbarSearch(e)"
-        @chiToolbarColumnsChange="(e) => chiToolbarColumnsChange(e)"
-        @chiToolbarColumnsReset="(e) => chiToolbarColumnsReset(e)"
-        @chiToolbarViewsChange="(e) => chiToolbarViewsChange(e)"
-      >
+      <DataTableToolbar>
```
Comment (SergioQCL) [inline src/chi-vue/src/views/DataTable/ClientSide/DataTableClientView.vue:60]: ❤️

File: src/chi-vue/src/views/DataTable/ClientSide/DataTableClientView.vue [hunk 10]
```diff
@@ -96,10 +90,7 @@
         <template v-slot:end>
           <div class="chi-toolbar__actions-desktop">
             <DownloadButtonIcon />
-            <ColumnCustomization
-              @chiColumnsReset="(e) => chiToolbarColumnsReset(e)"
-              :columnsData="toolbar.columnsData"
-            />
+            <ColumnCustomization @chiColumnsReset="(e) => chiColumnsReset(e)" :columnsData="toolbar.columnsData" />
```
Comment (SergioQCL) [inline src/chi-vue/src/views/DataTable/ClientSide/DataTableClientView.vue:93]: I think this works too. Can you please check?
```suggestion
            <ColumnCustomization @chiColumnsReset="chiColumnsReset" :columnsData="toolbar.columnsData" />
```

**Review by SergioQCL — APPROVED** [review]
image

**Review by jllr — APPROVED** [review]

**Pull request merged**